main.js

```diff
@@ -0,0 +1,220 @@
+// Set current date
+const date = new Date();
+const options = { weekday: 'long'};
+const formattedDate = new Intl.DateTimeFormat('en-US', options).format(date);
+
+
+const formattedDates = [];
+
+for (let i = 0; i < 5; i++) {
+    const nextDay = new Date(date);
+    nextDay.setDate(date.getDate() + i + 1);
+    const formattedDate = new Intl.DateTimeFormat('en-US', options).format(nextDay);
+    formattedDates.push(formattedDate); 
+}
+
+// Get current weather
+
+var currentWeather = [];
+
+var weatherForecast = [];
+
+document.querySelector('.search').addEventListener('click', function () {
+  var searchLocation = document.querySelector('#search-query').value;
+  var words = searchLocation.split(' ');
+  var city = words.join('/');
+
+  document.querySelector('#search-query').value = '';
+  currentWeather = [];
+  weatherForecast = [];
+
+  fetchLocation(city);
+  fetchPlace(city);
+});
+
+var fetchLocation = function(city) {
+  const apiKey = '0e79198cac257d0139d13e84e8617759'
+  const url = `http://api.openweathermap.org/geo/1.0/direct?q=${city}&appid=` + apiKey;
+  fetch(url, {
+    method: 'GET',
+    dataType: 'json'
+  })
+  .then(data => data.json())
+  .then(data => fetchWeather(data));
+};
+
+var fetchWeather = function(data) {
+  const apiKey = '0e79198cac257d0139d13e84e8617759'
+  const url = `https://api.openweathermap.org/data/2.5/weather?lat=${data[0].lat}&lon=${data[0].lon}&appid=` + apiKey + `&units=imperial`;
+  fetch(url, {
+    method: 'GET',
+    dataType: 'json'
+  })
+  .then(data => data.json())
+  .then(data => addWeather(data));
+};
+
+var addWeather = function (data) {
+  currentWeather.push({
+    temp: Math.round(data.main.temp),
+    name: data.name,
+    conditions: data.weather[0].main,
+    icon: `https://openweathermap.org/img/wn/${data.weather[0].icon}@4x.png`
+  })
+
+  renderWeather();
+};
+
+var renderWeather = function() {
+  document.querySelector('.current-weather').replaceChildren();
+
+  for (let i = 0; i < currentWeather.length; i++) {
+    const weather = currentWeather[i];
+
+    var template = `
+    <div class='col-md-4'>
+      <div class='weather'>
+        <h1>${weather.temp}º</h1>
+        <h2>${weather.name}</h2>
+        <h3>${weather.conditions}</h3>
+      </div>
+    </div>
+    <div class='col-md-4'>
+      <div class='icon'>
+        <img src=${weather.icon}>
+      </div>
+    </div>`;
+
+    document.querySelector('.current-weather').insertAdjacentHTML('beforeend', template);
+  }
+};
+
+// Get 5 day forecast
+
+var fetchPlace = function(city) {
+  const url = 'http://api.openweathermap.org/geo/1.0/direct?q=';
+  const apiKey = '&appid=0e79198cac257d0139d13e84e8617759'
+  const fullUrl = url + city + apiKey;
+  fetch(fullUrl, {
+    method: 'GET',
+    dataType: 'json'
+  })
+  .then(data => data.json())
+  .then(data => fetchForecast(data));
+};
```

Comment:
You are doing the same 3 times, you could have a generic function that takes api, url and callback for the data and reuse it. 
Important DRY principle. 
```suggestion
const fetchPlace = (fullUrl, callback) => {
  fetch(fullUrl, {
    method: 'GET',
    dataType: 'json'
  })
  .then(data => data.json())
  .then(data => callBack(data));
};
```

Reply:
Way more readable!!!